### ISSUE TITLE
Add more about monetization to Getting Paid guide

### ISSUE THREAD

**Issue by nayafia** [github/opensource.guide#357]
In an email convo with @isaacs, he wondered why we didn't explicitly discuss the VC route in ["Getting Paid for Open Source Work"](https://opensource.guide/getting-paid/).

I intentionally didn't touch on business models because I think it only applies to a small percentage of potential readers' situations. I wanted this guide to be more broad and answer the question, _"How do I get paid for my time (directly or indirectly)?"_ rather than _"How do projects make money?"_

Many open source contributors aren't involved in the project's overall direction. And not all projects are suitable for monetization.

That being said, I'm happy to write up a section if there's interest.

Common business models to cover:

* Dual licensing (example: Sidekiq / @mperham)
* Enterprise (example: npm / @isaacs)
* SaaS (example: Ghost / @JohnONolan and @ErisDS)

Any thoughts/feedback welcomed!

**Comment by theo-armour** [comment]
@nayafia 

Monetizing start-up open source projects is a fascinating field of thought - and we are just at the beginning of the possibilities.

But I agree that:

> it only applies to a small percentage of potential readers' situations

What the 'Getting Paid for Open Source Work' does not mention is that much - maybe even most - open source work is carried out by salaried peeps.

Think of:

* Linux, BSD, Chrome OS and all the open source operating systems
* Python, PHP, JavaScript, Fortran and all the languages
* Git, Subversion and all the revision control systems
* All the libraries, IDEs and tools to support the above
* W3C, Khronos Goup and all the people writing specs and building tests
* And much more

Tens of thousands - maybe even millions - of people help develop open source software and get paid handsomely for doing so while working at large and small companies, universities government and elsewhere.

Maybe this won't help the little start-up with a genius idea, but there are a huge number of opportunities where starting my making small contributions may lead to a well-paid, fully professional career.
 
It might be a nice idea if the 'Getting paid...' section pointed out some of the great career opportunities and not just start-up possibilities.

**Comment by isaacs** [comment]
I agree with @theo-armour, the number 1 way that OSS contributors pay their bills is with full-time programming jobs, and much (most?) of OSS contributions are made on the clock.

**Comment by isaacs** [comment]
(sorry premature submit)

The reason why I brought up VC in our email conversation is that, in the context of foundations, kickstarters, and other "self-funding" routes, VC is a pretty common and worthwhile avenue to explore.  But really, the vast majority of OSS projects never crowdfund, join a foundation, or create a corporation.  They remain side (or main) projects of people working for a company.

**Comment by nayafia** [comment]
@theo-armour oh interesting. I agree, that was the intent behind [this section](https://opensource.guide/getting-paid/#funding-your-own-time):

> Today, many people get paid to work part- or full-time on open source. The most common way to get paid for your time is to talk to your employer.

My goal with this guide was to frame it more in the way you describe (getting paid on the clock), and less about funded projects .

Do you have suggestions on how I could make that section more explicit? Is it more about being hired based on your open source work, vs. asking your current employer?

@isaacs ah, that makes sense! I'll add a bullet to that section about VC.

**Comment by theo-armour** [comment]
@nayafia 

> Getting Paid for Open Source Work
> Sustain your work in open source by getting financial support for your time or your project.

I don't want to complicate things too much but it's a really interesting, fraught and complicated topic.

This morning I wrote down my response to the challenge. I hope, somehow, this is helpful.

Theo

***
# Getting Paid for Open Source Work
_Sustain your work in open source by getting financial support for your time or your project._

There are at least two aspects to making some money with open source software development:

* The first is becoming visible to communities who would be willing to pay you for your efforts
* The second is the growing of communities that have the resources to pay you

## Becoming Visible
An artist creates a painting. Somebody buys the painting.
An author writes a story. Somebody publishes it.

Selling work product is very old stuff. And yet the process keeps getting more sophisticated and there are more opportunities to check out the work prior to making a decision.

Today a graphic designer shows a web page of logos with links to the customers sites.

A musician builds a web page with links to the songs they have recorded

Clients and customers have many more opportunities to try before they buy.

Similar things are happening with software development.

In the old days the code you wrote was hidden in some inaccessible proprietary binary file.

Today the situation for programmers has improved because of open source.

Anecdotally, in my field of 3D and WebGL, I know that [Brandon Jones]( https://github.com/toji ) was hired by Google because of his [gl-matrix]( https://github.com/toji/gl-matrix ) library. [Aki Rodic]( https://github.com/arodic ) (now in Shanghai ) was first hired by Google because of his [jellyfish]( http://arodic.github.io/p/jellyfish/ ) script.

Of course, you cannot see the code people write that is still embedded in proprietary apps, but increasing numbers of peeps have contributions visible in open source efforts.


##  Resourceful Communities
Let's look at some actual numbers.

These days we can look at the contributions of millions of programmers. The GitHub API is one of several great resources for this work.

My current tool is [GubGub]( https://pushme-pullyou.github.io/gubgub/r2/index.html ) a very work-in-progress script that uses the GitHub API to find interesting things in public, open-source repositories. The default search identifies the top 100 repositories with the most stars.

The list is displayed in the 'Select GitHub user and repo' in the side menu.

It's fairly easy to identify different types of successful open-source communities:

* Spin-off projects from established companies - Bootstrap, React, Angular
* Independent projects that grew into communities and started getting support - D3, jQuery, FontAwesome
* Projects maintained by individuals that are funded in a variety of ways - Linux, Animate.css, Three.js

There are also an increasing number of teaching or support projects where students are encouraged to star projects as part of the lesson plan.

All of these projects are large enough that they employ people or people are employed because of their good skill and knowledge in using the project.

None or almost none of these projects were directly funded by VCs or angels.

There is a good mix of corporate spin-off off projects and individual hacker projects.

Even smaller projects have dozens of contributors. At the top there's room for many people.

There's nobody gunning for an IPO here. There are no Zuckerbergs, Bezos, Jobs or Gates here - just large communities of moderately prosperous people doing nice, constructive things in a collaborative ways.

## Conclusions
There are two way of making money:

Contribute to large projects and become good enough so that people pay you.

Start something new by yourself or with a few people and grow it until enough people benefit from your work so they want to pay for more development and support.

The really fun thing is doing a bit of both - learning helping others and using what you learn to start your own new efforts.

**Comment by lutzhorn** [comment]
Why a new paragraph for every short sentence? The structure of your argument is hard to follow in this list of small sentences that read like a list of bullet points without bullets.

**Comment by theo-armour** [comment]
@lutzhorn 

Nice question. ;-)

It's definitely a curious style that I have evolved.

It comes out of a fondness for:

* Bulleted lists
* Narrow newspaper columns
* Vertical scrolling being so easy

My coding style is similar. It's very open and airy.

It follows [Mr.doob's Code Style™]( https://github.com/mrdoob/three.js/wiki/Mr.doob's-Code-Style%E2%84%A2 ) 

I work a lot with his [Three.js]( https://three.js.org ) library, so it made sense to follow his style.

It was quite painful at first.

**Comment by lutzhorn** [comment]
Well, code style is nothing I would use for text that tries to descibe something or makes an argument. I find your text above very hard to read because I can't see what belongs together and where a new thought or description of a fact begins.

**Comment by nayafia** [comment]
@theo-armour thanks for typing up these thoughts, I enjoyed reading them!

The "Getting Paid" guide is currently structured into two parts: 1) getting paid for OSS time as a salaried employee, and 2) finding independent funding for your project. I'm going to keep that structure, but if there are parts you think we can augment based on what you've shared here, feel free to open an issue or PR so we can discuss. (I think it'd make a great blog post as well if you wanted to publish them elsewhere 😊)

Closing this issue in favor of #361.

**Comment by Aparus** [comment]
I wandered that in "Getting payed" section there is nothing about 'double licensing' when it is one of the first things mentioned in 'Starting an OS project'. 
I have translated 3 sections into Russian... and I hope did deep dive into material... 
So controversial staff... So many logical conflicts between 'getting-started' and 'getting-payed'.
